### PR TITLE
Update translation of `Array.includes()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

更新 array.includes 翻译
### Motivation

同上
### Additional details

无

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 title: Array.prototype.includes()
 slug: Web/JavaScript/Reference/Global_Objects/Array/includes
+page-type: javascript-instance-method
+browser-compat: javascript.builtins.Array.includes
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:5]:
```suggestion
```
remove according to [guide](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#%E5%85%83%E6%95%B0%E6%8D%AE)

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 2]
```diff
@@ -19,70 +21,89 @@ includes(searchElement, fromIndex)
...
-
-    > **备注：** 使用 `includes()` 比较字符串和字符时是区分大小写的。
-
+  - : 需要查找的值。
 - `fromIndex` {{optional_inline}}
-  - : 从`fromIndex` 索引处开始查找 `searchElement`。如果为负值，则按升序从 `array.length + fromIndex` 的索引开始搜（即使从末尾开始往前跳 `fromIndex` 的绝对值个索引，然后往后搜寻）。默认为 0。
+  - ：从零开始搜索的索引，[转换为整数](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
+    - 负索引从数组末尾开始计数 —— 假如 `fromIndex < 0` ， `fromIndex + array.length` 被调用。然而，在这种情况下，仍然从前到后搜索数组。
+    - 如果 `fromIndex < -array.length` 或者省略 `fromIndex` ，则使用 `0` ，导致搜索整个数组。
+    - 如果 `fromIndex >= array.length` ，则不会搜索数组并返回 `false` 。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:29]:
```suggestion
    - 负索引从数组末尾开始计数——假如 `fromIndex < 0`，`fromIndex + array.length` 被调用。然而，在这种情况下，仍然从前到后搜索数组。
    - 如果 `fromIndex < -array.length` 或者省略 `fromIndex`，则使用 `0`，导致搜索整个数组。
    - 如果 `fromIndex >= array.length`，则不会搜索数组并返回 `false`。
```
remove spaces

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 3]
```diff
@@ -19,70 +21,89 @@ includes(searchElement, fromIndex)
...
 
 ### 返回值
 
-返回一个布尔值 {{jsxref("Boolean")}} 。
-如果在数组中（或 `fromIndex` 指定的范围中）找到了 `searchElement`，则返回 `true`，否则返回 `false`。
+一个布尔值，如果在数组中找到 `searchElement` 值，则该值为 `true` （或者在指定的索引 `fromIndex` 所指示的数组部分中找到 `searchElement` 值）。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:33]:
```suggestion
一个布尔值，如果在数组中找到 `searchElement` 值，则该值为 `true`（或者在指定的索引 `fromIndex` 所指示的数组部分中找到 `searchElement` 值）。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 4]
```diff
@@ -19,70 +21,89 @@ includes(searchElement, fromIndex)
...
+
+## 描述
 
-0 的值将全部视为相等，与符号无关（即 -0 与 0 和 +0 相等），但 `false` 不被认为与 0 相等。
+`includes()` 方法使用[SameValueZero](/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality)算法将 `searchElement` 与数组中的元素进行比较。0的值都被认为是相等的，不管符号是什么。（即 `-0` 等于 `0` ），但 `false` 不被认为与 `0` 相同。[`NaN`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN)可以被正确搜索。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:37]:
```suggestion
`includes()` 方法使用 [SameValueZero](/zh-CN/docs/Web/JavaScript/Reference/Operators/Strict_equality)算法将 `searchElement` 与数组中的元素进行比较。0 值都被认为是相等的，不管符号是什么。（即 `-0` 等于 `0`），但 `false` 不被认为与 `0` 相同。[`NaN`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/NaN) 可以被正确搜索。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 5]
```diff
@@ -19,70 +21,89 @@ includes(searchElement, fromIndex)
...
-arr.includes('c', 3);   // false
-arr.includes('c', 100); // false
+arr.includes("c", 3); // false
+arr.includes("c", 100); // false
 ```
 
 ### 计算出的索引小于 0
 
-如果 `fromIndex` 为负值，计算出的索引将作为开始搜索`searchElement`的位置。如果计算出的索引小于 0，则整个数组都会被搜索。
+如果 `fromIndex` 为负值，计算出的索引将作为开始搜索 `searchElement` 的位置。如果计算出的索引小于 `0` ，则整个数组都会被搜索。
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:69]:
```suggestion
如果 `fromIndex` 为负值，计算出的索引将作为开始搜索 `searchElement` 的位置。如果计算出的索引小于 `0`，则整个数组都会被搜索。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 6]
```diff
@@ -96,8 +117,10 @@ arr.includes('a', -2); // false
 ## 参见
 
 - [Polyfill of `Array.prototype.includes` in `core-js`](https://github.com/zloirock/core-js#ecmascript-array)
```
Comment (coolzjy) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:119]:
```suggestion
- [`core-js` 中 `Array.prototype.includes` 的 polyfill](https://github.com/zloirock/core-js#ecmascript-array)
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 7]
```diff
@@ -19,70 +21,89 @@ includes(searchElement, fromIndex)
...
 - `searchElement`
-
-  - : 需要查找的元素值。
-
-    > **备注：** 使用 `includes()` 比较字符串和字符时是区分大小写的。
-
+  - : 需要查找的值。
 - `fromIndex` {{optional_inline}}
-  - : 从`fromIndex` 索引处开始查找 `searchElement`。如果为负值，则按升序从 `array.length + fromIndex` 的索引开始搜（即使从末尾开始往前跳 `fromIndex` 的绝对值个索引，然后往后搜寻）。默认为 0。
+  - ：从零开始搜索的索引，[转换为整数](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
```
Comment (fwqaaq) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:26]:
```suggestion
  - ：从零开始搜索的索引，[转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 8]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
-
-  - : 需要查找的元素值。
-
-    > **备注：** 使用 `includes()` 比较字符串和字符时是区分大小写的。
-
+  - : 需要查找的值。
 - `fromIndex` {{optional_inline}}
-  - : 从`fromIndex` 索引处开始查找 `searchElement`。如果为负值，则按升序从 `array.length + fromIndex` 的索引开始搜（即使从末尾开始往前跳 `fromIndex` 的绝对值个索引，然后往后搜寻）。默认为 0。
+  - ：从零开始搜索的索引，[转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
+    - 负索引从数组末尾开始计数——假如 `fromIndex < 0`，`fromIndex + array.length` 被调用。然而，在这种情况下，仍然从前到后搜索数组。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:25]:
```suggestion
    - 负索引从数组末尾开始计数——如果 `fromIndex < 0`，那么实际使用的是 `fromIndex + array.length`。然而在这种情况下，数组搜索仍然从前往后。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 9]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
+    - 如果 `fromIndex < -array.length` 或者省略 `fromIndex`，则使用 `0`，导致搜索整个数组。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:26]:
```suggestion
    - 如果 `fromIndex < -array.length` 或者省略 `fromIndex`，则使用 `0`，导致整个数组被搜索。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 10]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
+    - 如果 `fromIndex >= array.length`，则不会搜索数组并返回 `false`。
 
 ### 返回值
 
-返回一个布尔值 {{jsxref("Boolean")}} 。
-如果在数组中（或 `fromIndex` 指定的范围中）找到了 `searchElement`，则返回 `true`，否则返回 `false`。
+一个布尔值，如果在数组中找到 `searchElement` 值，则该值为 `true`（或者在指定的索引 `fromIndex` 所指示的数组部分中找到 `searchElement` 值）。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:31]:
```suggestion
一个布尔值，如果在数组中（如果 `fromIndex` 给定，那么在它所指示的数组部分中）找到 `searchElement` 值，则该值为 `true`。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 11]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
+
+## 描述
 
-0 的值将全部视为相等，与符号无关（即 -0 与 0 和 +0 相等），但 `false` 不被认为与 0 相等。
+`includes()` 方法使用[零值相等](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#零值相等)算法将 `searchElement` 与数组中的元素进行比较。0 值都被认为是相等的，不管符号是什么。（即 `-0` 等于 `0`），但 `false` 不被认为与 `0` 相同。[`NaN`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/NaN) 可以被正确搜索。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:35]:
```suggestion
`includes()` 方法使用[零值相等](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#零值相等)算法将 `searchElement` 与数组中的元素进行比较。0 值都被认为是相等的，不管符号是什么。（即 `-0` 等于 `0`），但 `false` 不被认为与 `0` 相同。[`NaN`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/NaN) 可以被正确搜索到。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 12]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
 
-> **备注：** 技术上来讲，`includes()` 使用 [`零值相等`](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#%E9%9B%B6%E5%80%BC%E7%9B%B8%E7%AD%89) 算法来确定是否找到给定的元素。
+当使用[稀疏数组](/zh-cn/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)时， `includes()` 方法迭代空槽，就像它们的值 `undefined` 一样。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:37]:
```suggestion
当使用[稀疏数组](/zh-cn/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)时， `includes()` 方法迭代空槽，就像它们的值是 `undefined` 一样。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 13]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
+
+`includes()` 方法是[通用的](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/Array#通用的数组方法)。它只期望 `this` 值具有 `length` 属性和整型键属性。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:39]:
```suggestion
`includes()` 方法是[通用的](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/Array#通用的数组方法)。它只期望 `this` 值具有 `length` 属性和整数键属性。
```

Heads up: this note is present on almost every array method page. Translators will want to make sure that they use the same message format for every page.

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 14]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
+[1, 2, 3].includes(2); // true
+[1, 2, 3].includes(4); // false
+[1, 2, 3].includes(3, 3); // false
 [1, 2, 3].includes(3, -1); // true
 [1, 2, NaN].includes(NaN); // true
+["1", "2", "3"].includes(3); // false
 ```
 
-### fromIndex 大于等于数组长度
+### fromIndex 大于或等于数组长度
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:54]:
```suggestion
### fromIndex 大于等于数组长度
```

Original is better; "greater than or equal to" always translates to 「大于等于」

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 15]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
 
-如果 `fromIndex` 大于等于数组的长度，则将直接返回 `false`，且不搜索该数组。
+如果 `fromIndex` 大于或者等于数组的长度，则将直接返回 `false`，且不搜索该数组。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:56]:
```suggestion
如果 `fromIndex` 大于等于数组的长度，则将直接返回 `false`，且不搜索该数组。
```

**Review by Josh-Cena — COMMENTED** [review]
Random drive-by. I wrote these pages so I do have some sense of Chinese translations in mind 😄

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 16]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
 - `searchElement`
-
-  - : 需要查找的元素值。
-
-    > **备注：** 使用 `includes()` 比较字符串和字符时是区分大小写的。
-
+  - : 需要查找的值。
 - `fromIndex` {{optional_inline}}
-  - : 从`fromIndex` 索引处开始查找 `searchElement`。如果为负值，则按升序从 `array.length + fromIndex` 的索引开始搜（即使从末尾开始往前跳 `fromIndex` 的绝对值个索引，然后往后搜寻）。默认为 0。
+  - ：从零开始搜索的索引，[转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:24]:
```suggestion
  - : 开始搜索的索引（从零开始），[会转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#转换为整数)。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 17]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
 
 ### 计算出的索引小于 0
 
-如果 `fromIndex` 为负值，计算出的索引将作为开始搜索`searchElement`的位置。如果计算出的索引小于 0，则整个数组都会被搜索。
+如果 `fromIndex` 为负值，计算出的索引将作为开始搜索 `searchElement` 的位置。如果计算出的索引小于 `0`，则整个数组都会被搜索。
 
 ```js
 // array length is 3
 // fromIndex is -100
 // computed index is 3 + (-100) = -97
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:72]: 翻译一下注释

```suggestion
// 数组长度为 3
// fromIndex 为 -100
// 计算出的索引为 3 + (-100) = -97
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 18]
```diff
@@ -19,70 +19,89 @@ includes(searchElement, fromIndex)
...
+arr.includes("b", -100); // true
+arr.includes("c", -100); // true
+arr.includes("a", -2); // false
 ```
 
-### 作为通用方法的 includes()
+### 对稀疏数组使用 includes() 方法
 
-`includes()` 方法有意设计为通用方法。它不要求`this`值是数组对象，所以它可以被用于其他类型的对象 (比如类数组对象)。下面的例子展示了 在函数的 [arguments](/zh-CN/docs/Web/JavaScript/Reference/Functions/arguments) 对象上调用的 `includes()` 方法。
+你可以在稀疏数组中搜索 `undefined` ，得到 `true` 。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:84]:
```suggestion
你可以在稀疏数组中搜索 `undefined`，得到 `true` 。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md [hunk 19]
```diff
@@ -95,9 +114,11 @@ arr.includes('a', -2); // false
 
 ## 参见
 
-- [Polyfill of `Array.prototype.includes` in `core-js`](https://github.com/zloirock/core-js#ecmascript-array)
-- {{jsxref("TypedArray.prototype.includes()")}}
-- {{jsxref("String.prototype.includes()")}}
+- [`core-js` 中 `Array.prototype.includes` 的 polyfill](https://github.com/zloirock/core-js#ecmascript-array)
+- [Indexed collections](/en-US/docs/Web/JavaScript/Guide/Indexed_collections)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/includes/index.md:118]: 指向中文文档

```suggestion
- [索引集合类](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections)
```

**Review by yin1999 — APPROVED** [review]
:+1:

**Pull request merged**